### PR TITLE
Add tests for String.prototype.startsWith

### PR DESCRIPTION
**Pull request by leobalter**

### CODE REVIEW

File: test/built-ins/String/prototype/startsWith/coerced-values-of-position.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+assert(str.startsWith('The future', NaN), 'NaN coerced to 0');
+assert(str.startsWith('The future', null), 'null coerced to 0');
+assert(str.startsWith('The future', false), 'false coerced to 0');
+assert(str.startsWith('The future', ''), '"" coerced to 0');
+assert(str.startsWith('The future', '0'), '"0" coerced to 0');
+assert(str.startsWith('The future', undefined), 'undefined coerced to 0');
+assert(str.startsWith('The future', 0.4), '0.4 coerced to 0');
+
+assert(str.startsWith('The future', -1));
```
Comment (jugglinmike) [inline test/built-ins/String/prototype/startsWith/coerced-values-of-position.js]: This assertion is for integer boundary enforcement (not integer coercion), so I think it belongs in another file

File: test/built-ins/String/prototype/startsWith/return-false-with-out-of-bounds-position.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  str.startsWith('!', 100), false,
+  'str.startsWith("!", 100) returns false'
+);
+
+assert.sameValue(
+  str.startsWith('!', Infinity), false,
+  'str.startsWith("!", Infinity) returns false'
+);
+
+assert.sameValue(
```
Comment (jugglinmike) [inline test/built-ins/String/prototype/startsWith/return-false-with-out-of-bounds-position.js]: This assertion is repeated

File: test/built-ins/String/prototype/startsWith/return-false-with-out-of-bounds-position.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  str.startsWith('!', str.length), false,
+  'str.startsWith("!", str.length) returns false'
+);
```
Comment (jugglinmike) [inline test/built-ins/String/prototype/startsWith/return-false-with-out-of-bounds-position.js]: We should also have tests for successful matches when the position is out of bounds. Consider renaming this file and including them here

File: test/built-ins/String/prototype/startsWith/searchstring-found-without-position.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  15. If the sequence of elements of S starting at start of length searchLength
+  is the same as the full element sequence of searchStr, return true.
+  ...
+---*/
+
+var str = 'The future is cool!';
+
+assert(str.startsWith('The '), 'str.startsWith("The ") === true');
+assert(str.startsWith('The future'), 'str.startsWith("The future") === true');
+assert(str.startsWith(str), 'str.startsWith(str) === true');
```
Comment (jugglinmike) [inline test/built-ins/String/prototype/startsWith/searchstring-found-without-position.js]: Good idea!

**Comment by jugglinmike** [comment]
Just a few ideas for changes here

**Comment by leobalter** [comment]
done! Thanks for reviewing.

**Comment by bterlson** [comment]
Looks good. :eggplant:

**Comment by leobalter** [comment]
aliased `squash` to `eggplant`, rebased and :eggplant:ed

**Comment by bterlson** [comment]
Thanks!

**Pull request merged**